vault/auth.go

```diff
@@ -406,45 +406,49 @@ func (c *Core) persistAuth(ctx context.Context, table *MountTable, localOnly boo
 		}
 	}
 
-	if !localOnly {
-		// Marshal the table
-		compressedBytes, err := jsonutil.EncodeJSONAndCompress(nonLocalAuth, nil)
+	writeTable := func(mt *MountTable, path string) error {
```

Comment:
There's a very similar function in `persistMounts` (same except for the log message), could we breaking this out to a method that can be share in both places?